components/net/fetch/methods.rs

```diff
@@ -530,6 +530,15 @@ pub async fn main_fetch(
         let should_replace_with_mixed_content = !response_is_network_error &&
             should_response_be_blocked_as_mixed_content(request, &response, &context.protocols);
 
+        let (mut should_replace_with_csp_error, mut violations) = (false, vec![]);
+        if !response.is_network_error() {
+            (should_replace_with_csp_error, violations) =
+                should_response_be_blocked_by_csp(request, &response.actual_response());
+        }
+        if !violations.is_empty() {
+            target.process_csp_violations(request, violations);
+        }
```

Comment:
The other checks do one of two alternatives:

1. Return a `bool` and do the reporting of violations inside the method
2. Return the checkresult and check against that

I am in favor of 1 here, since that hides a lot of the complexity and makes it clearer that CSP stuff is handled inside of `should_response_be_blocked_by_csp` and doesn't leak outside of it. That also avoids the need for `mut`, since it would handle that logic inside of the method.

E.g. move `response.is_network_error()` and `!violations.is_empty()` into `should_response_be_blocked_by_csp` so that this becomes

```suggestion
        let (should_replace_with_csp_error, violations) = should_response_be_blocked_by_csp(request, &response);
```

Reply:
This is much nicer.